src/mcode-pilot/services/outcomes/CLQOutcomesService.js

```diff
@@ -0,0 +1,209 @@
+    constructor(params) {
+        super();
+        this.serviceUrl = params.serviceUrl;
+        this.timescale = params.timescale || [];
+    }
+
+    /*
+    Based off of the mcode namespace find the elements that match it in the filter
+    */
+    findMcodeElements(props, mcodeType, single = false) {
+        let found = [];
+        _.forIn(props, function (value, key) {
+            if (value.mcodeElement === mcodeType && value.selected) {
+                found.push(value);
+            }
+        });
+        return single ? found[0] : found;
+    }
+
+    /* Build the CLQ demograpchics filter section based off of the Compass filter criteria
+     */
+    buildDemographicsFilter(similarPatientProps) {
+        let demoProps = similarPatientProps.demographic.options;
+        let filter = {};
+        let gender = this.findMcodeElements(demoProps, "shr.core.BirthSex", true);
+        let race = this.findMcodeElements(demoProps, "shr.core.Race", true);
+        let age = this.findMcodeElements(demoProps, "shr.core.DateOfBirth", true);
+        let age_at_diagnosis = this.findMcodeElements(demoProps, "shr.core.DateOfDiagnosis", true);
+        if (gender) {
+            filter.gender = {
+                codeSystemName: "AdministrativeGender",
+                codeSystem: "2.16.840.1.113883.4.642.2.1",
+                code: gender.value
+            };
+        }
+        if (age) {
+            filter.age = {
+                min: age.minValue,
+                max: age.maxValue
+            };
+        }
+        if (age_at_diagnosis) {
+            filter.age_at_diagnosis = {
+                min: age_at_diagnosis.minValue,
+                max: age_at_diagnosis.maxValue
+            };
+        }
+        if (race) {
+            filter.race = {
+                "codeSystemName": "HL7 v3 Code System Race",
+                "codeSystem": "2.16.840.1.113883.5.104",
+                "code": race.value
+            };
+        }
+        return filter;
+    }
+
+    /* Build the CLQ diagnosis filter based off of the Comapss Filter options
+     */
+    buildDiagnosisFilter(similarPatientProps) {
+        let pathologyProps = similarPatientProps.pathology.options;
+        let filter = {};
+        let stage = this.findMcodeElements(pathologyProps, "onco.core.TNMClinicalStageGroup", true);
+        let t = this.findMcodeElements(pathologyProps, "onco.core.TNMClinicalPrimaryTumorCategory", true);
+        let n = this.findMcodeElements(pathologyProps, "onco.core.TNMClinicalRegionalNodesCategory", true);
+        let m = this.findMcodeElements(pathologyProps, "onco.core.TNMClinicalDistantMetastasesCategory", true);
+        let grade = this.findMcodeElements(pathologyProps, "onco.core.CancerHistologicGrade", true);
+        if (stage) {
+            filter.stage = stage.reference.stage;
+        }
+        if (grade) {
+            filter.grade = grade.reference.getGradeAsSimpleNumber();
+        }
+        filter.tmn = {};
```

Comment:
The sample request json attached to Jira doesn't include this part, but should this be `tmn` or `tnm`? TNM is what I've generally seen used elsewhere, ex `onco.core.TNMClinicalStageGroup`

Reply:
Change made, just waiting for travis.